// mat-file-upload/esm5/lib/mat-file-upload.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatInputModule } from '@angular/material/input';
import { MatSelectModule } from '@angular/material/select';
import { MatFileUploadComponent } from './mat-file-upload.component';
import * as i0 from "@angular/core";
var MatFileUploadModule = /** @class */ (function () {
    function MatFileUploadModule() {
    }
    MatFileUploadModule.ɵmod = i0.ɵɵdefineNgModule({ type: MatFileUploadModule });
    MatFileUploadModule.ɵinj = i0.ɵɵdefineInjector({ factory: function MatFileUploadModule_Factory(t) { return new (t || MatFileUploadModule)(); }, imports: [[
                CommonModule,
                MatButtonModule,
                MatIconModule,
                MatInputModule,
                MatSelectModule,
                FormsModule,
            ]] });
    return MatFileUploadModule;
}());
export { MatFileUploadModule };
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(MatFileUploadModule, { declarations: [MatFileUploadComponent], imports: [CommonModule,
        MatButtonModule,
        MatIconModule,
        MatInputModule,
        MatSelectModule,
        FormsModule], exports: [MatFileUploadComponent] }); })();
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(MatFileUploadModule, [{
        type: NgModule,
        args: [{
                declarations: [MatFileUploadComponent],
                imports: [
                    CommonModule,
                    MatButtonModule,
                    MatIconModule,
                    MatInputModule,
                    MatSelectModule,
                    FormsModule,
                ],
                exports: [MatFileUploadComponent],
            }]
    }], null, null); })();
